.github/actions/install-mysql/action.yml

```diff
@@ -0,0 +1,21 @@
+name: MySQL installer  
+description: 'Install a mysql instance for the tests to run'
+jobs:
+
+runs:
+  using: "composite"  
+  steps:
+    - name: Environment variables
+      env:        
+        CI: true
+        RAILS_ENV: test
+        RACK_ENV: test
+        DISABLE_SPRING: 1
+      run: echo "Environment variables were set up for MySQL"
+      shell: bash    
```

Comment:
I'm not particularly familiar with GitHub actions, but my interpretation here is that this will set the specified environment variables, but only for this one job; according to https://docs.github.com/en/actions/learn-github-actions/environment-variables#passing-values-between-steps-and-jobs-in-a-workflow, we need to write these values to the GITHUB_ENV environment file for them to be used in subsequent steps.

Is that accurate, or is there something I'm missing?

Reply:
You are right! I actually have a new version where I moved them up

I spent so much time trying to find this issue and yes, they are only shared by steps. So far I have identified 3 scopes: By job, by step and by action.